version: "3"

networks:
  gitea:
    external: false

services:
  server:
    image: gitea/gitea:latest
    container_name: gitea
    environment:
      - USER_UID=1000
      - USER_GID=1000
      - GITEA__database__DB_TYPE=postgres
      - GITEA__database__HOST=db:5432
      - GITEA__database__NAME=gitea
      - GITEA__database__USER=gitea
      - GITEA__database__PASSWD=gitea
    restart: always
    networks:
      - gitea
    volumes:
      - ./gitea:/data
      - /etc/timezone:/etc/timezone:ro
      - /etc/localtime:/etc/localtime:ro
    ports:
      - "3000:3000"
      - "222:22"
    depends_on:
      - db

  db:
    image: postgres:14
    restart: always
    environment:
      - POSTGRES_USER=gitea
      - POSTGRES_PASSWORD=gitea
      - POSTGRES_DB=gitea
    networks:
      - gitea
    volumes:
      - ./postgres:/var/lib/postgresql/data
  ui:
    # image: joeljosephgeorge/gitea-ui:v1.0.3
    build:
      context: ../UI
      dockerfile: Dockerfile
    healthcheck:
      timeout: 45s
      interval: 10s
      retries: 10
    container_name: gitea-ui

    # env_file: 
    #   - .env.local
    environment:
      - VITE_GITEA_BASE_URL=http://localhost:3000
      - VITE_GITEA_UI_URL=http://localhost:3001
      - VITE_GITEA_ORG_NAME=BCS
      - VITE_FASTAPI_BASE_URL=http://localhost:8000
      #The below values are dummy values, replace them with your own values
      #the values where not taking when running the api and ui in docker, i could run it by adding 
      #them in the .env.local file and adding the file in the env_file section
      - VITE_GITEA_CLIENT_ID = {your-client-id}
      - VITE_GITEA_CLIENT_SECRET = {your-client-secret}
      - VITE_GITEA_REDIRECT_URI=http://localhost:3001/oauth/callback

    restart: always
    networks:
      - gitea
    ports:
      - "3001:3001"
    depends_on:
      - server
